### PR TITLE
VEP-554: address comments

### PR DESCRIPTION
**Pull request by mivanov1988**
This PR aims to address the comments from 
https://github.com/vmware/versatile-data-kit/pull/748.

Signed-off-by: Miroslav Ivanov miroslavi@vmware.com

### CODE REVIEW

File: specs/vep-554-apache-airflow-integration/README.MD [hunk 1]
```diff
@@ -356,15 +298,86 @@ with DAG(
...
 ### Security and Permissions
 
 The communication between VDK Provider and Control Service will be secured through the
 use of HTTPS and OAuth 2.0. The most commonly used OAuth 2.0 flows will be supported (
 e.g. Client Credentials Flow, Refresh Token Flow, etc.).
 
-Secrets will be stored as part
-of [Airflow Connection](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html)
-like follows:
+Secrets will be stored as part of Airflow Connection like follows:
```
Comment (gageorgiev) [inline specs/vep-554-apache-airflow-integration/README.MD:380]:
```suggestion
Secrets will be stored as part of the Airflow Connection like follows:
```

**Review by gageorgiev — APPROVED** [review]

File: specs/vep-554-apache-airflow-integration/README.MD [hunk 2]
```diff
@@ -187,58 +173,14 @@ by providing a sensor operator derived from `airflow.sensors.base.BaseSensorOper
...
-        * Mitigations: What can be done to stop the bleeding, especially for already
-          running user workloads?
-        * Diagnostics: What are the useful log messages and their required logging levels
-          that could help debug the issue?
-        * Testing: Are there any tests for failure mode? If not, describe why._
-* Operability
-    * What are the SLIs (Service Level Indicators) an operator can use to determine the
-      health of the system.
-    * What are the expected SLOs
-* Test Plan
```
Comment (tozka) [inline specs/vep-554-apache-airflow-integration/README.MD:224]: I am not seeing a test plan . 

Particularly are there any recommendations and practices that we follow for adding new provider. If there's link of Airflow doc, add it here. Or do we need to come up with something ourselves?

File: specs/vep-554-apache-airflow-integration/README.MD [hunk 3]
```diff
@@ -356,15 +298,86 @@ with DAG(
...
+
+During the first 4 hours of the workflow execution, the Provider will perform about 6 (
+3 jobs * 2 requests/minute) requests per minute or 1440 requests in total. For the next 2
+hours the Provider will perform about 4 (2 jobs * 2 requests/minute) requests per minute
+or 440 requests in total. In total the Provider will perform 1880 requests for the whole
+workflow execution.
+
+### Troubleshooting
+
+In order to provide better traceability, the Data Jobs within the same workflow will have
```
[the remark this answers is not in the document]
Comment (tozka) [reply specs/vep-554-apache-airflow-integration/README.MD:339]: Also. 

I (data engineer/data scientist) develop 3 data jobs and schedule them in Airflow. 

Would I be using Airflow UI to see the status of the jobs? How would I see why it failed (user error, patlfrom error- > next level details then memory error, bug error, syntax error, which step) . How can I see the logs of the job? 

How can I remediate the issue (restart it from Airflow UI, repdeloy the job ? , revert the job ?). 
How can I monitor for those issues?

File: specs/vep-554-apache-airflow-integration/README.MD [hunk 4]
```diff
@@ -356,15 +298,86 @@ with DAG(
...
+* `Job 4` depends on `Job 1` and `Job 2`
+* `Job 5` depends on `Job 3`
+* The execution time of each of the following jobs is about 4 hours: `Job 1`, `Job 2`
+  , and `Job 3`
+* The execution time of each of the following jobs is about 2 hours: `Job 4`
+  and `Job 5`
+
+![sample-workflow.png](sample-workflow.png)
+
+During the first 4 hours of the workflow execution, the Provider will perform about 6 (
```
[the remark this answers is not in the document]
Comment (mivanov1988) [reply specs/vep-554-apache-airflow-integration/README.MD:331]: We should do proper measurements before recommending any. Also, I don't think the recommended ones are precise.

File: specs/vep-554-apache-airflow-integration/README.MD (hunk 2 above)
Comment (mivanov1988) [reply specs/vep-554-apache-airflow-integration/README.MD:224]: We will follow the official Airflow documentation - https://airflow.apache.org/docs/apache-airflow-providers/howto/create-update-providers.html#initial-code-and-unit-tests

**Pull request merged**